dask/dataframe/core.py

```diff
@@ -578,6 +578,61 @@ def clear_divisions(self):
         divisions = (None,) * (self.npartitions + 1)
         return type(self)(self.dask, self._name, self._meta, divisions)
 
+    def calculate_divisions(self, col=None):
+        """Calculate the divisions for the dataframe.
```

Comment:
```suggestion
    def compute_divisions(self, col=None):
        """Compute the current divisions for the dataframe.
```
`compute` seems like more consistent terminology.

Also we should clearly distinguish between "what are the current divisions of this DataFrame (when they're unknown)", and "what are good divisions to use with this DataFrame if I'm going to reindex by `col`"? This method seems like the former; I think we'll want a `compute_good_divisions(col, npartitions=None)` as well (and maybe call this `compute_current_divisions`?)

Reply:
Yes I was totally waffling on the name. I'm happy with compute. We could also consider something like `get_divisions` or `ddf.divisions.compute()`, or `compute_known_divisions`?

Reply:
`ddf.divisions.compute()` would be nice, but would mean `ddf.divisions` would have to be some sort of delayed collection, which would break a lot of things. I still like `compute_current_divisions` the most; `compute_known_divisions` feels contradictory to `ddf.known_divisions` to me.

Reply:
Yeah that works for me.